File: src/main/java/ru/arc/invest/items/SimpleItem.java
package ru.arc.invest.items;

import lombok.AllArgsConstructor;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

@AllArgsConstructor

public class SimpleItem extends GenericItem {

    Material material;

    public static SimpleItem deserialize(String type){
        return new SimpleItem(Material.matchMaterial(type.toUpperCase()));
    }

    public static SimpleItem fromMaterial(String s){
        Material material = Material.matchMaterial(s.toUpperCase());
        if(material == null) return null;
        return new SimpleItem(material);
    }

    @Override
    public String toString() {
        return "SimpleItem{" +
                "material=" + material +
                '}';
    }

    @Override
    public ItemStack stack() {
        return new ItemStack(material);
    }

    @Override
    public boolean fits(ItemStack stack) {
        return this.material == stack.getType();
    }
}

File: src/main/kotlin/ru/arc/invest/items/SimpleItem.kt
package ru.arc.invest.items

import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class SimpleItem(private val material: Material) : GenericItem() {

    override fun stack(): ItemStack = ItemStack(material)

    override fun fits(stack: ItemStack): Boolean = material == stack.type

    override fun toString(): String = "SimpleItem{material=$material}"

    companion object {
        @JvmStatic
        fun deserialize(type: String): SimpleItem = SimpleItem(Material.matchMaterial(type.uppercase())!!)

        @JvmStatic
        fun fromMaterial(s: String): SimpleItem? {
            val material = Material.matchMaterial(s.uppercase()) ?: return null
            return SimpleItem(material)
        }
    }
}
